### PR TITLE
[Gluten-167] Fallback ShuffleExchangeExec with RangePartitioning for ClickHouse backend.

### LINKED ISSUE #167: Fallback ShuffleExchangeExec with RangePartitioning for ClickHouse backend.
Currently, ClickHouse backend doesn't support ShuffleExchangeExec with RangePartitioning, so it needs to fallback.

**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
(This section helps developers understand the context and *why* for this feature, in addition to  the *what*)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zzcclp**
Fallback ShuffleExchangeExec with RangePartitioning for ClickHouse backend.

Close #167 .

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: jvm/src/main/scala/io/glutenproject/execution/HashAggregateExecTransformer.scala [hunk 1]
```diff
@@ -415,11 +415,19 @@ case class HashAggregateExecTransformer(
       val childrenNodeList = new util.ArrayList[ExpressionNode]()
       val childrenNodes = aggExpr.mode match {
         case Partial =>
-          aggregatFunc.children.toList.map(expr => {
-            val aggExpr: Expression = ExpressionConverter
-              .replaceWithExpressionTransformer(expr, originalInputAttributes)
-            aggExpr.asInstanceOf[ExpressionTransformer].doTransform(args)
-          })
+          if (!GlutenConfig.getConf.isClickHouseBackend) {
```
Comment (rui-mo) [inline jvm/src/main/scala/io/glutenproject/execution/HashAggregateExecTransformer.scala:418]: Will we move the code related to certain backend only to that specific module in the future?

**Comment by rui-mo** [comment]
Verified on Jenkins and locally.

File: jvm/src/main/scala/io/glutenproject/execution/HashAggregateExecTransformer.scala (hunk 1 above)
Comment (zzcclp) [reply jvm/src/main/scala/io/glutenproject/execution/HashAggregateExecTransformer.scala:418]: yes, I will optimize these logic codes later.

**Comment by marin-ma** [comment]
LGTM.

**Comment by rui-mo** [comment]
Verifed arrow backend on Jenkins.

**Pull request merged**